dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -448,74 +454,209 @@ public void Test_Incremental_Job_Test_Pages_Are_Found_Create_And_Publish_New_Pag
+         */
+
+        TestCaseSiteSearch case3 = new TestCaseSiteSearchBuilder()
+                .defaultContentToDefaultLanguage(true)
+                .defaultPageToDefaultLanguage(true)
+                .siteSearchSecondLanguage(true)
+                .createContentInSecondLanguage(true)
+                .createPageInDefaultLanguage(true)
+                .expectedResultsWhenSearchingContentInSecondLanguage(new ExpectedResults(true, false))
+                .createTestCaseSiteSearch();
+
+
+        /*
+         * Same as above but with DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=false
+         * Expected: searching for content in second language should give NO results.
+         */
+
+        TestCaseSiteSearch case4 = new TestCaseSiteSearchBuilder()
+                .defaultContentToDefaultLanguage(true)
+                .defaultPageToDefaultLanguage(false)
+                .siteSearchSecondLanguage(true)
+                .createContentInSecondLanguage(true)
+                .createPageInDefaultLanguage(true)
+                .expectedResultsWhenSearchingContentInSecondLanguage(new ExpectedResults(false, false))
+                .createTestCaseSiteSearch();
+
+        /*
+         * Given sceneario: two-language content referenced from a page in default lang only. DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=true.
+         * Create a Site-Search run including both languages
+         * Expected: searching content of either version of the content should give results. The result should be the page in the default language
+         */
+
+        TestCaseSiteSearch case5 = new TestCaseSiteSearchBuilder()
+                .defaultContentToDefaultLanguage(true)
+                .defaultPageToDefaultLanguage(true)
+                .siteSearchDefaultLanguage(true)
+                .siteSearchSecondLanguage(true)
+                .createContentInDefaultLanguage(true)
+                .createContentInSecondLanguage(true)
+                .createPageInDefaultLanguage(true)
+                .expectedResultsWhenSearchingContentInDefaultLanguage(new ExpectedResults(true, false))
+                .expectedResultsWhenSearchingContentInSecondLanguage(new ExpectedResults(true, false))
+                .createTestCaseSiteSearch();
+
+
+
+        return new TestCaseSiteSearch[] {case1, case2, case2_1, case3, case4, case5};
+    }
+
+    @UseDataProvider("siteSearchTestCases")
     @Test
-    public void test_MultilangContent_IndexingAllLanguages()
+    public void testSiteSearchDifferentScenarios(final TestCaseSiteSearch testCase)
             throws DotPublishingException, JobExecutionException, DotDataException, IOException, DotSecurityException, WebAssetException {
 
         boolean defaultContentToDefaultLangOriginalValue =
                 Config.getBooleanProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE", false);
 
-        try {
+        boolean defaultPagetoDefaultLangOriginalValue =
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'defaultPagetoDefaultLangOriginalValue' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)